host_core/native/hostcore_wasmcloud_native/src/lib.rs

```diff
@@ -70,10 +75,123 @@ rustler::init!(
         par_cache_path,
         detect_core_host_labels,
         pk_from_seed,
+        get_provider_bindle,
+        get_actor_bindle,
     ],
     load = load
 );
 
+#[rustler::nif]
```

Comment:
Retrieving bindles should be on the dirtyIO scheduler. Elixir has a really really short timeout on NIF executions on the default scheduler. The "dirty IO" scheduler allows it to run on a background thread and not cause a NIF crash.

```rust
#[rustler::nif(schedule = "DirtyIo")]
```